java/openmldb-jdbc/src/main/java/com/_4paradigm/openmldb/sdk/impl/InsertPreparedStatementImpl.java

```diff
@@ -300,22 +311,26 @@ public void setObject(int i, Object o, int i1) throws SQLException {
 
     private void buildRow() throws SQLException {
         SQLInsertRow currentRow = getSQLInsertRow();
-
         boolean ok = currentRow.Init(stringsLen);
         if (!ok) {
             throw new SQLException("init row failed");
         }
 
-        for (int i = 0; i < currentDatasType.size(); i++) {
-            Object data = currentDatas.get(i);
+        // SQLInsertRow::AppendXXX order is the schema order(skip the no-hole columns)
```

Comment:
`addBatch` will call `buildRow`. so it will sort multi times

Reply:
ok